packages/ui/dev/scripts/generators/generateWorkers.ts

```diff
@@ -1,38 +1,94 @@
 import faker from 'faker'
 
+import { Opening } from './generateOpeningsAndUpcomingOpenings'
 import { WORKING_GROUPS } from './generateWorkingGroups'
 import { Mocks } from './types'
-import { randomUniqueArrayFromRange, randomFromRange } from './utils'
+import { randomFromRange, randomUniqueArrayFromRange } from './utils'
 
 let nextId = 0
 
-const generateWorker = (type: string, groupId: number) => (memberId: number) => ({
-  id: String(nextId++),
-  membershipId: memberId,
-  groupId: groupId,
-  applicationId: randomFromRange(0, 100),
-  status: type,
-  rewardPerBlock: randomFromRange(1, 2) * 100,
-  earnedTotal: randomFromRange(10, 40) * 100,
-  missingRewardAmount: randomFromRange(0, 20) * 100,
-  stake: randomFromRange(3, 8) * 1000,
-  nextPaymentAt: faker.date.soon(randomFromRange(1, 10)).toJSON(),
-})
+const generateWorker = (type: string, groupId: string, applications: Application[], opening?: Opening) => (
+  memberId: number
+) => {
+  if (!opening) {
+    return
+  }
+  const runtimeId = nextId++
+
+  const application = extracted(opening, 'filled')(String(memberId))
+  applications.push(application)
+
+  return {
+    id: `${groupId}-${runtimeId}`,
+    runtimeId,
+    membershipId: memberId,
+    groupId,
+    applicationId: application.id,
+    status: type,
+    rewardPerBlock: randomFromRange(1, 2) * 100,
+    earnedTotal: randomFromRange(10, 40) * 100,
+    missingRewardAmount: randomFromRange(0, 20) * 100,
+    stake: randomFromRange(3, 8) * 1000,
+    nextPaymentAt: faker.date.soon(randomFromRange(1, 10)).toJSON(),
+  }
+}
 
 export type Worker = ReturnType<ReturnType<typeof generateWorker>>
 
+let nextApplicationId = 0
+function extracted(opening: Opening, status = 'pending') {
```

Comment:
Not crucial, but here should be `generateApplication` instead of `extracted`

Reply:
Thanks! I forgot about that :facepalm: